sumomongodbatlascollector/api.py

```diff
@@ -92,11 +101,14 @@ def fetch(self):
         next_request = True
         count = 0
         sess = ClientMixin.get_new_session()
-        self.log.info(f'''Fetching LogType: {log_type}  starttime: {kwargs['params']['minDate']} endtime: {kwargs['params']['maxDate']}''')
+        self.log.info(f'''Fetching LogType: {log_type}  starttime: {kwargs['params']['minDate']} endtime: {kwargs['params']['maxDate']} url: {url}''')
         try:
             while next_request:
                 send_success = has_next_page = False
+                start_time = time.time()
                 status, data = ClientMixin.make_request(url, method="get", session=sess, logger=self.log, TIMEOUT=self.collection_config['TIMEOUT'], MAX_RETRY=self.collection_config['MAX_RETRY'], BACKOFF_FACTOR=self.collection_config['BACKOFF_FACTOR'], **kwargs)
+                end_time = time.time()
+                self.log.info(f'''Fetching LogType: {log_type}  starttime: {kwargs['params']['minDate']} endtime: {kwargs['params']['maxDate']} url: {url} execution time: {end_time - start_time:.2f} seconds current_process_memory_usage: {self._get_current_process_memory_usage()}''')
```

Comment:
lets add current_process_memory_usage execution time for line 121 and 135 basically to track how much time it took to send to sumo

Reply:
Moved to the context manager based logging